### ISSUE TITLE
Add versioning to k8gb.io documentation

### ISSUE THREAD

**Issue by somaritane** [k8gb-io/k8gb#794]
As our releasing cadence increases, and we bring more breaking changes,
it might be useful to have a versioning system for our documentation.

This might help us to keep track of what's new in each release, especially when it comes to k8gb configuration values.

P.S. This might require changing of Jekyll theme to something that supports versioning,
or changing to some other static web-page authoring framework, which has that versioning support built-in, like [Hugo](https://gohugo.io/)

Assignees:
Labels: documentation

**Comment by jkremser** [comment]
there is a [docsy](https://github.com/google/docsy) hugo template used by a lot of cncf projects:
https://www.docsy.dev/docs/examples/ and also by k8s itself

**Comment by somaritane** [comment]
need to check how it supports versioning

**Comment by jkremser** [comment]
I think the version support is not there in the template or hugo itself.. well it can be done using the directories, like:
- `content/en/docs/v0.0.1/something.md`
- `content/en/docs/v0.0.2/something.md`

But this is not a good approach, imho. What I've seen is this:
1. have the main template for the landing page in 1 branch (be it `gh-pages` or anything else)
2. have the version specific docs stored in a separate branch using some template, like `v0.x-docs` (often this scheme corresponds with the strategy when breaking changes are introduced -> need for another parallel docs)
3. use the netlify feature for this: https://docs.netlify.com/domains-https/custom-domains/multiple-domains/#branch-subdomains together with auto builds
4. link those versioned docs from the landing page from some combo box

examples using this approach are:
- https://www.kubeflow.org/
- https://docs.k8ssandra.io/
- https://kubernetes.io/
(check the links when switching the versions)

**Comment by somaritane** [comment]
Found this in docsy's documentation:
https://www.docsy.dev/docs/adding-content/versioning/

**Comment by jkremser** [comment]
ah cool, yes, that looks like 4) from my description. So the combo box (/dropdown) is a built-in feature in the template, but they are not opinionated about the deployment of those versions

**Comment by somaritane** [comment]
yep, so we can use branch subdomains for publishing and then refer to these subdomains

**Comment by ytsarev** [comment]
@itsfarhan looks like something in your area of expertise :)

**Comment by itsfarhan** [comment]
@ytsarev Yes, currently we don't have versioning in our docs. I will implement this and create a PR.
